pkg/kubelet/container_gc.go

```diff
@@ -157,6 +170,18 @@ func (cgc *realContainerGC) GarbageCollect() error {
 		}
 	}
 
+	// Remove dead symlinks - should only happen on upgrade
+	// from a k8s version without proper log symlink cleanup
+	logSymlinks, _ := filepath.Glob(path.Join(cgc.containerLogsDir, "*.log"))
+	for _, logSymlink := range logSymlinks {
+		if _, err = os.Stat(logSymlink); os.IsNotExist(err) {
```

Comment:
ah sorry I didn't know about stat cmd line tool. anyway, os.Stat follows symlinks so this is why it works. os.Lstat would be the equivalent of what you did with stat.